src/fhost_fpga.py

```diff
@@ -150,36 +157,51 @@ def __init__(self, host, katcp_port=7147, boffile=None, connect=True, config=Non
             self.min_load_time = None
             self.network_latency_adjust = None
 
+        self._pfb_of_counts = {}
+        self._ct_counts = {}
+        self._qdr_counts = {}
+        self.fnums = int(configd['fengine']['f_per_fpga'])
+
     @classmethod
     def from_config_source(cls, hostname, katcp_port, config_source):
         boffile = config_source['bitstream']
         return cls(hostname, katcp_port=katcp_port, boffile=boffile,
                    connect=True, config=config_source)
 
-    def ct_okay(self, sleeptime=1):
+    def ct_okay(self, wait_time=1):
         """
-        Is the corner turner working?
-        :return: True or False,
+        Checks if the corner turner working.
+        :param: wait_time : Float or None : If not None, fetch ct counter,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is only cached if wait_time is None.
+        :return: bool : True, if corner tuner is okay
+
         """
-        ct_ctrs0 = self.registers.ct_ctrs.read()['data']
-        time.sleep(sleeptime)
-        ct_ctrs1 = self.registers.ct_ctrs.read()['data']
+        for cnt in range(self.fnums):
+            if wait_time is not None:
+                ct_ctrs_err0 = self.registers.ct_ctrs.read()['data']
+                time.sleep(wait_time)
+            else:
+                ct_ctrs_err0 = self._ct_counts.get(cnt, 0)
```

Comment:
From the frying pan into the fire... The default value should be a dict with the expected keys, but....

I don't think this code is correct if I compare it to the original. You're checking that the whole datastructure is equal, whereas the original code only checked certain fields. Also, there is only one register, the counts are incorporated in the field names. Perhaps we should have a chat about this?